shards/shards.go

```diff
@@ -439,17 +438,30 @@ func (ss *shardedSearcher) streamSearch(ctx context.Context, q query.Q, opts *zo
 
 	defer cancel()
 
+	g, ctx := errgroup.WithContext(childCtx)
+
 	// For each query, throttle the number of parallel
 	// actions. Since searching is mostly CPU bound, we limit the
 	// number of parallel searches. This reduces the peak working
 	// set, which hopefully stops https://cs.bazel.build from crashing
 	// when looking for the string "com".
-	feeder := make(chan zoekt.Searcher, len(shards))
-	for _, s := range shards {
-		feeder <- s
-	}
-	close(feeder)
-	g, ctx := errgroup.WithContext(childCtx)
+	feeder := make(chan zoekt.Searcher, runtime.GOMAXPROCS(0))
+	g.Go(func() error {
+		defer close(feeder)
+		for _, s := range shards {
+			if err := proc.Yield(ctx); err != nil {
```

Comment:
We chatted live about this. Yeah this is unnecessary since it is so fast. Will mention the more subtle point that once we do yield we will have up to `#workers` running + up to `cap(feeder)` shards (so `#cpus` * 2). So when we do yield, we may keep running for a short while. This is done for simplicity. The "better" place to add yield is in the workers, but then that adds in the need for concurrency control.